### PR TITLE
Swap out JSA-Content endpoint URL to avoid redirect

### PR DESCRIPTION
**Pull request by sherakama**
# Ready for review

### CODE REVIEW

**Comment by jbickar** [comment]
22 commits? `man sed`.

**Comment by sherakama** [comment]
The UI was open for me already.

**Comment by jbickar** [comment]
Do we need to update the occurrences of `https://sites.stanford.edu/jsa-content` in the various `ImporterFieldProcessorCustomBody.php` classes? See https://github.com/SU-SWS/stanford_install_tasks/blob/jsacontent/Stanford/Jumpstart/Install/Content/Importer/ImporterFieldProcessorCustomBody.php#L23

**Comment by jbickar** [comment]
```
DN0a24a13a: ~/Documents/working/stanford_install_tasks$ grep -r "sites.stanford.edu/jsa-content" .
./Stanford/Jumpstart/Install/Content/Importer/ImporterFieldProcessorCustomBody.php:        $entity->{$field_name}[LANGUAGE_NONE][0]['value'] = str_replace('/jsa-content', 'https://sites.stanford.edu/jsa-content', $entity->{$field_name}[LANGUAGE_NONE][0]['value']);
./Stanford/Jumpstart/Install/Content/Importer/ImporterFieldProcessorCustomBody.php:        $entity->{$field_name}[LANGUAGE_NONE][0]['summary'] = str_replace('/jsa-content', 'https://sites.stanford.edu/jsa-content', $entity->{$field_name}[LANGUAGE_NONE][0]['summary']);
./Stanford/Jumpstart/Install/Content/Importer/ImporterFieldProcessorCustomBody.php:        $entity->{$field_name}[LANGUAGE_NONE][0]['safe_value'] = str_replace('/jsa-content', 'https://sites.stanford.edu/jsa-content', $entity->{$field_name}[LANGUAGE_NONE][0]['safe_value']);
./Stanford/Jumpstart/Install/Content/Importer/ImporterFieldProcessorCustomBody.php:        $entity->{$field_name}[LANGUAGE_NONE][0]['safe_summary'] = str_replace('/jsa-content', 'https://sites.stanford.edu/jsa-content', $entity->{$field_name}[LANGUAGE_NONE][0]['safe_summary']);
./Stanford/DrupalCamp/Install/Content/Importer/ImporterFieldProcessorCustomBody.php:        $entity->{$field_name}[LANGUAGE_NONE][0]['value'] = str_replace('/jsa-content', 'https://sites.stanford.edu/jsa-content', $entity->{$field_name}[LANGUAGE_NONE][0]['value']);
./Stanford/DrupalCamp/Install/Content/Importer/ImporterFieldProcessorCustomBody.php:        $entity->{$field_name}[LANGUAGE_NONE][0]['summary'] = str_replace('/jsa-content', 'https://sites.stanford.edu/jsa-content', $entity->{$field_name}[LANGUAGE_NONE][0]['summary']);
./Stanford/DrupalCamp/Install/Content/Importer/ImporterFieldProcessorCustomBody.php:        $entity->{$field_name}[LANGUAGE_NONE][0]['safe_value'] = str_replace('/jsa-content', 'https://sites.stanford.edu/jsa-content', $entity->{$field_name}[LANGUAGE_NONE][0]['safe_value']);
./Stanford/DrupalCamp/Install/Content/Importer/ImporterFieldProcessorCustomBody.php:        $entity->{$field_name}[LANGUAGE_NONE][0]['safe_summary'] = str_replace('/jsa-content', 'https://sites.stanford.edu/jsa-content', $entity->{$field_name}[LANGUAGE_NONE][0]['safe_summary']);
./Stanford/JumpstartLab/Install/PathologicSettings.php:      'https://sites.stanford.edu/jsa-content/',
```

**Comment by sherakama** [comment]
All those do is make an attempt at replacing the absolute path to file in the body field. Old content will still need the old path but new content may need the new path. 

As we are not actively creating new content for these install profiles I would say that we are probably ok to punt on that for now.

**Comment by jbickar** [comment]
Is it not replacing `/jsa-content` _with_ `https://sites.stanford.edu/jsa-content`?

**Comment by sherakama** [comment]
Sorry, you are correct I had it backwards. Should be ok to leave those as is.

**Comment by jbickar** [comment]
🤔

**Comment by jbickar** [comment]
In for a penny, in for a pound. Might as well change them now.

**Comment by sherakama** [comment]
Please dont, they need to stay put. We have content that has `/jsa-content/path-to-file.img` in the body fields.

**Comment by jbickar** [comment]
OK, I'm misunderstanding what this is doing. Is this not changing that to `https://sites.stanford.edu/jsa-content/path-to-file.img`?

**Comment by sherakama** [comment]
Oh, I see what you did there. I guess that is fine. We have a big mis-mash-up of urls on that site. I guess one more isn't a big deal.

**Review by jbickar — APPROVED** [review]
Sure hope we didn't have any typoes.

**Comment by jbickar** [comment]
Those camelCase CodeClimate errors - is that an issue with our `.codeclimate.yml` file?

**Comment by sherakama** [comment]
Yeah, we can probably fix that. CamelCase is not required there.

**Comment by jbickar** [comment]
```
      Drupal NamingConventions ValidFunctionName ScopeNotCamelCaps:
enabled: false
```
and
```
      Drupal NamingConventions ValidFunctionName NotCamelCaps:
enabled: false
```
?

**Comment by sherakama** [comment]
Yeah. that looks like it
https://github.com/SU-SWS/stanford_module_example/blob/8.x-1.x/.codeclimate.yml#L48

**Comment by sherakama** [comment]
This is not a Drupal project specifically so it did not get all of the standards.

**Comment by jbickar** [comment]
Yeah, that's what I was just thinking. I don't want to add all the Drupals.

**Comment by sherakama** [comment]
I lied 
```
Controversial/CamelCaseParameterName:
        enabled: false
```

**Comment by sherakama** [comment]
screen shot 2018-12-11 at 12 58 47 pm
screen shot 2018-12-11 at 12 58 50 pm

File: .codeclimate.yml [hunk 1]
```diff
@@ -8,11 +8,18 @@ engines:
     checks:
       Drupal Commenting DocComment ShortNotCapital:
         enabled: false
+      Drupal NamingConventions ValidFunctionName NotCamelCaps:
```
Comment (jbickar) [inline .codeclimate.yml:11]: @sherakama do you want these in here as well, or take them out now that `Controversial/CamelCaseParameterName` is disabled?

Comment (sherakama) [reply .codeclimate.yml:11]: Leave em. They are in our default config anyhow.

Comment (jbickar) [reply .codeclimate.yml:11]: **_FINE._**

**Pull request merged**